pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SingleSnapshotAbortedTxnProcessorImpl.java

```diff
@@ -98,13 +101,20 @@ public CompletableFuture<PositionImpl> recoverFromSnapshot() {
                                 }
                             }
                         }
-                        closeReader(reader);
                         return CompletableFuture.completedFuture(startReadCursorPosition);
+                    } catch (NullPointerException npe) {
```

Comment:
Could we check if the message is null or not directly?

Reply:
Already fixed, thanks